File: CallApiProject/src/Models/Menu.ts
import { Level } from "./Level";
import { Meal } from "./Meal";

export class Menu {
    constructor(
        public menuCode: number,
        public menuName: string,
        public discription: string,
        public userCode: number,
        public dateCreated: Date,
        public dateUpdated: Date,
        public links: string,
        public viewsNumber: number,
        public userName: string, 
        public dateUpload:Date,
        public pictureCode:number,        
        public publish:boolean,
        public levelCode:number,
        public meals:Meal[],
        )
                 { }
}

File: MenuCalculator/DAL/Models/Menu.cs
using System;
using System.Collections.Generic;

namespace DAL.Models
{
    public partial class Menu
    {
        public Menu()
        {
            CategoriesToMenu = new HashSet<CategoriesToMenu>();
            Meals = new HashSet<Meal>();
        }

        public int MenuCode { get; set; }
        public string MenuName { get; set; }
        public string Discription { get; set; }
        public int? UserCode { get; set; }
        public DateTime? DateCreated { get; set; }
        public DateTime? DateUpdated { get; set; }
        public string Links { get; set; }
        public int? ViewsNumber { get; set; }
        public DateTime? DateUpload { get; set; }
        public int? PictureCode { get; set; }
        public bool? Publish { get; set; }
        public int? LevelCode { get; set; }

        public virtual Users User { get; set; }
        public virtual Level LevelCodeNavigation { get; set; }
        public virtual Picture PictureCodeNavigation { get; set; }
        public virtual ICollection<Meal> Meals { get; set; }
        public virtual ICollection<CategoriesToMenu> CategoriesToMenu { get; set; }
    }
}
